app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt

```diff
@@ -27,6 +33,46 @@ class FractionInteractionViewModel(
     return userAnswerBuilder.build()
   }
 
+  override fun isExplicitErrorCheckRequired(): Boolean = true
+
+  override fun getPendingAnswerError(category: AnswerErrorCategory): String? {
+    if (category == AnswerErrorCategory.REAL_TIME) {
+      return if (answerText.isNotEmpty() && StringToFractionParser().getRealTimeError(answerText.toString()) != FractionParsingError.VALID)
```

Comment:
You mean you want me to create fractionParser() inside FractionInteractionViewModel instead of StringToFractionParser? and why we need parser here?

Reply:
@BenHenning PTAL on above question

Reply:
No sorry, what I mean is we should have a ``val`` field that is the ``StringToFractionParser`` so that we only construct it once, and reuse it. There's no benefit to creating a bunch of ``StringToFractionParser`` objects in this model class.

Reply:
Okay